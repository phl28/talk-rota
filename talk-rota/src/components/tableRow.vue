<script setup lang="ts">
import { ref, watch } from 'vue'
import dayjs from 'dayjs';
import { KInputDate, KButton } from "@cambridgekineticsltd/kinetic-ui";

const props = defineProps({
    rowData: Object,
    index: Number,
    changeDateBool: Function,
    chgDate: Boolean,
    changeDate: Function,
})

  const dateValue = ref(dayjs(props.rowData?.date))


watch(dateValue, (newDate) => {
  if (props.changeDate) {
    props.changeDate(props.index, newDate)
  }
})
</script> 

<template>
  <tr>
    <td>{{ rowData?.name }}</td>
    <td>{{ rowData?.date  }}</td>
    <td><KButton @click="changeDateBool" size="sm" variant="primary" label="Change Date"/></td>
    <td><k-input-date v-if="chgDate" v-model="dateValue" label="Date"/></td>
  </tr>
  <!-- @k-input-date="changeDate" -->
</template> 

<style>
.table th,
.table td {
  padding: 10px;
  text-align: left;
}

.table th {
  background-color: rgb(45, 10, 48);
  font-weight: bold;
}

.table tr:nth-child(even) {
  background-color: rgb(54, 15, 41);
}
</style>