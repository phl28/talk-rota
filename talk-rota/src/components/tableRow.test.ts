import { mount } from "@vue/test-utils";
import tableRow from "./tableRow.vue";
import { test, expect } from "vitest";

test("mount component", async () => {
    expect(tableRow).toBeTruthy();

    const wrapper = mount(tableRow, {
        props: {
            rowData: {
                name: "test",
                date: "2021-01-01",
            }
        }
    })
    expect(wrapper.html()).toMatchSnapshot();

});