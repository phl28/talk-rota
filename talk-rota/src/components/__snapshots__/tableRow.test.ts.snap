// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `
"<tr>
  <td>test</td>
  <td>2021-01-01</td>
  <td><button type=\\"button\\" class=\\"btn btn btn-sm btn-primary\\" aria-disabled=\\"false\\" aria-pressed=\\"false\\" title=\\"Change Date\\">
      <!----> Change Date
      <!---->
    </button></td>
  <td>
    <!--v-if-->
  </td>
</tr>
<!-- @k-input-date=\\"changeDate\\" -->"
`;
